api/v1beta1/openstacknetattachment_webhook.go

```diff
@@ -0,0 +1,114 @@
+func (r *OpenStackNetAttachment) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/mutate-osp-director-openstack-org-v1beta1-openstacknetattachment,mutating=true,failurePolicy=fail,sideEffects=None,groups=osp-director.openstack.org,resources=openstacknetattachments,verbs=create;update,versions=v1beta1,name=mopenstacknetattachment.kb.io,admissionReviewVersions={v1,v1beta1}
+
+var _ webhook.Defaulter = &OpenStackNetAttachment{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *OpenStackNetAttachment) Default() {
+	openstacknetattachmentlog.Info("default", "name", r.Name)
+
+	// To avoid validation errors, we need to set the DesiredState to {} if it is nil.
+	// This is the error that occurs otherwise...
+	//
+	// spec.attachConfiguration.nodeNetworkConfigurationPolicy.desiredState:
+	// Invalid value: "null": spec.attachConfiguration.nodeNetworkConfigurationPolicy.desiredState in body must be of type object: "null"
+	//
+	if r.Spec.AttachConfiguration.NodeNetworkConfigurationPolicy.DesiredState.Raw == nil {
+		r.Spec.AttachConfiguration.NodeNetworkConfigurationPolicy.DesiredState = nmstateshared.NewState("{}")
+	}
+
+}
+
+//+kubebuilder:webhook:path=/validate-osp-director-openstack-org-v1beta1-openstacknetattachment,mutating=false,failurePolicy=fail,sideEffects=None,groups=osp-director.openstack.org,resources=openstacknetattachments,verbs=create;update,versions=v1beta1,name=vopenstacknetattachment.kb.io,admissionReviewVersions={v1,v1beta1}
+
+var _ webhook.Validator = &OpenStackNetAttachment{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *OpenStackNetAttachment) ValidateCreate() error {
+	openstacknetattachmentlog.Info("validate create", "name", r.Name)
+
+	return nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *OpenStackNetAttachment) ValidateUpdate(old runtime.Object) error {
+	openstacknetattachmentlog.Info("validate update", "name", r.Name)
+
+	return r.checkBridgeName(old)
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *OpenStackNetAttachment) ValidateDelete() error {
+	openstacknetattachmentlog.Info("validate delete", "name", r.Name)
+
+	return nil
+}
+
+func (r *OpenStackNetAttachment) checkBridgeName(old runtime.Object) error {
+
+	// Get the current (potentially new) bridge name, if any
+	curBridge, err := nmstate.GetDesiredStatedBridgeName(r.Spec.AttachConfiguration.NodeNetworkConfigurationPolicy.DesiredState.Raw)
+
+	if err != nil {
+		return err
+	}
+
+	// Get the old bridge name, if any
+	var ok bool
+	var oldInstance *OpenStackNetAttachment
+
+	if oldInstance, ok = old.(*OpenStackNetAttachment); !ok {
+		return fmt.Errorf("runtime object is not an OpenStackNet")
```

Comment:
Hopefully this code is never reached, but I suppose `s/OpenStackNet/OpenStackNetAttachment/`

Reply:
thanks fixed with the latest added commit